api/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java

```diff
@@ -44,15 +47,24 @@
 import org.projectnessie.client.auth.NessieAuthenticationProvider;
 import org.projectnessie.client.http.v1api.HttpApiV1;
 import org.projectnessie.client.http.v2api.HttpApiV2;
+import org.projectnessie.client.rest.NessieHttpResponseFilter;
 import org.projectnessie.model.ser.Views;
 
 /**
  * A builder class that creates a {@link NessieHttpClient} via {@link HttpClientBuilder#builder()}.
  */
 public class HttpClientBuilder implements NessieClientBuilder<HttpClientBuilder> {
 
-  private final HttpClient.Builder builder = HttpClient.builder();
-  private HttpAuthentication authentication;
+  private static final ObjectMapper MAPPER =
+      new ObjectMapper()
+          .enable(SerializationFeature.INDENT_OUTPUT)
+          .disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
+
+  private final HttpClient.Builder builder =
+      HttpClient.builder()
+          .setObjectMapper(MAPPER)
+          .addResponseFilter(new NessieHttpResponseFilter());
```

Comment:
I wonder: we should probably not add this filter by default. For the OAuth2 client, it doesn't make sense for instance (it's not harmful either, just conceptually wrong since the client is not talking to Nessie at all).

Reply:
Yes, doesn't make sense for OAuth. Probably safer to not add it for OAuth and only for Nessie REST.

Reply:
Actually scratch that, the OAuth client uses `HttpClient.Builder`, not `HttpClientBuilder`... naming is hard :-)